### PR TITLE
Add Bamboo Block Mappings

### PR DESCRIPTION
**Pull request by AzaleeX**
## Introduction
Added mappings for Bamboo blocks, including Bamboo Door and Bamboo Sign, to the ItemSerializerDeserializerRegistrar.

> [!WARNING]
> I'm not sure it's legal to put IDSs directly in functions such as getLogIdentifier() or getAllSidedLogIdentifier() because banboo has nothing similar and has no log.

## Tests

I tested this PR by doing the following (tick all that apply):
- [ ] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [ ] Playtesting using a Minecraft client (provide screenshots or a video)
- [ ] Writing a test plugin (provide the code and sample output)
- [ ] Other (provide details)

### CODE REVIEW

File: src/block/WoodLikeBlockIdHelper.php [hunk 1]
```diff
@@ -88,9 +90,13 @@ public static function getSlabIdentifier(WoodType $type) : BID{
 			WoodType::CRIMSON => Ids::CRIMSON_SLAB,
 			WoodType::WARPED => Ids::WARPED_SLAB,
 			WoodType::CHERRY => Ids::CHERRY_SLAB,
+			WoodType::BAMBOO => Ids::BAMBOO_SLAB,
 		});
 	}
 
+	/**
+	 * @throws \Exception
+	 */
```
Comment (dktapps) [inline src/block/WoodLikeBlockIdHelper.php]: Unrelated change

File: src/block/WoodLikeBlockIdHelper.php [hunk 2]
```diff
@@ -103,6 +109,7 @@ public static function getLogIdentifier(WoodType $treeType) : BID{
 			WoodType::CRIMSON => Ids::CRIMSON_STEM,
 			WoodType::WARPED => Ids::WARPED_STEM,
 			WoodType::CHERRY => Ids::CHERRY_LOG,
+			WoodType::BAMBOO => 59000,
```
Comment (dktapps) [inline src/block/WoodLikeBlockIdHelper.php]: No

File: src/block/WoodLikeBlockIdHelper.php [hunk 3]
```diff
@@ -118,6 +125,7 @@ public static function getAllSidedLogIdentifier(WoodType $treeType) : BID{
 			WoodType::CRIMSON => Ids::CRIMSON_HYPHAE,
 			WoodType::WARPED => Ids::WARPED_HYPHAE,
 			WoodType::CHERRY => Ids::CHERRY_WOOD,
+			WoodType::BAMBOO => 59001,
```
Comment (dktapps) [inline src/block/WoodLikeBlockIdHelper.php]: Also no

File: src/data/bedrock/item/ItemSerializerDeserializerRegistrar.php [hunk 4]
```diff
@@ -136,6 +136,7 @@ private function register1to1BlockMappings() : void{
 		$this->map1to1Block(Ids::CAULDRON, Blocks::CAULDRON());
 		$this->map1to1Block(Ids::CHAIN, Blocks::CHAIN());
 		$this->map1to1Block(Ids::CHERRY_DOOR, Blocks::CHERRY_DOOR());
+		$this->map1to1Block(Ids::BAMBOO_DOOR, Blocks::BAMBOO_DOOR());
```
Comment (dktapps) [inline src/data/bedrock/item/ItemSerializerDeserializerRegistrar.php]: This list is supposed to be in alphabetical order

File: src/data/bedrock/item/ItemSerializerDeserializerRegistrar.php [hunk 5]
```diff
@@ -187,6 +188,7 @@ private function register1to1ItemMappings() : void{
 		$this->map1to1Item(Ids::CHAINMAIL_LEGGINGS, Items::CHAINMAIL_LEGGINGS());
 		$this->map1to1Item(Ids::CHARCOAL, Items::CHARCOAL());
 		$this->map1to1Item(Ids::CHERRY_SIGN, Items::CHERRY_SIGN());
+		$this->map1to1Item(Ids::BAMBOO_SIGN, Items::BAMBOO_SIGN());
```
Comment (dktapps) [inline src/data/bedrock/item/ItemSerializerDeserializerRegistrar.php]: Same again here

File: src/item/StringToItemParser.php [hunk 6]
```diff
@@ -147,6 +147,16 @@ private static function registerBlocks(self $result) : void{
 		$result->registerBlock("azure_bluet", fn() => Blocks::AZURE_BLUET());
 		$result->registerBlock("bamboo", fn() => Blocks::BAMBOO());
 		$result->registerBlock("bamboo_sapling", fn() => Blocks::BAMBOO_SAPLING());
```
Comment (dktapps) [inline src/item/StringToItemParser.php]: Alphabetical order

File: src/item/VanillaItems.php [hunk 7]
```diff
@@ -387,6 +388,7 @@ protected static function setup() : void{
 		self::register("carrot", new Carrot(new IID(Ids::CARROT), "Carrot"));
 		self::register("charcoal", new Coal(new IID(Ids::CHARCOAL), "Charcoal"));
 		self::register("cherry_sign", new ItemBlockWallOrFloor(new IID(Ids::CHERRY_SIGN), Blocks::CHERRY_SIGN(), Blocks::CHERRY_WALL_SIGN()));
+		self::register("bamboo_sign", new ItemBlockWallOrFloor(new IID(Ids::BAMBOO_SIGN), Blocks::BAMBOO_SIGN(), Blocks::BAMBOO_WALL_SIGN()));
```
Comment (dktapps) [inline src/item/VanillaItems.php]: Alphabetical order

File: src/item/VanillaItems.php [hunk 8]
```diff
@@ -331,6 +331,7 @@
  * @method static WritableBook WRITABLE_BOOK()
  * @method static WrittenBook WRITTEN_BOOK()
  * @method static SpawnEgg ZOMBIE_SPAWN_EGG()
+ * @method static ItemBlockWallOrFloor BAMBOO_SIGN()
```
Comment (dktapps) [inline src/item/VanillaItems.php]: This docblock is supposed to be generated by a script, not hand-edited

**Review by dktapps — CHANGES_REQUESTED** [review]
I think the reason we didn't already add these was because of the issue around logs, but this is not the right way to fix it.

File: src/block/WoodLikeBlockIdHelper.php (hunk 2 above)
Comment (AzaleeX) [reply src/block/WoodLikeBlockIdHelper.php]: What should I do?

File: src/block/WoodLikeBlockIdHelper.php (hunk 3 above)
Comment (AzaleeX) [reply src/block/WoodLikeBlockIdHelper.php]: What should I do?

File: src/block/WoodLikeBlockIdHelper.php (hunk 2 above)
Comment (nicholass003) [reply src/block/WoodLikeBlockIdHelper.php]: You have to update or create new ids here
https://github.com/pmmp/PocketMine-MP/blob/stable/src/block/BlockTypeIds.php

Comment (nicholass003) [reply src/block/WoodLikeBlockIdHelper.php]: Example :
https://github.com/pmmp/PocketMine-MP/pull/6391/files#diff-b7d3ed653a6892f8d17adae002eb6840e0d6ab5fba40e9a7afce040bd3ea5094

File: src/block/WoodLikeBlockIdHelper.php (hunk 3 above)
Comment (ClousCloud) [reply src/block/WoodLikeBlockIdHelper.php]: You have to add the id to the src/block/BlockTypeIds.php file

File: src/item/VanillaItems.php (hunk 8 above)
Comment (AzliRynz) [reply src/item/VanillaItems.php]: Yeah, he should run the `composer update-codegen` command

File: src/block/WoodLikeBlockIdHelper.php (hunk 3 above)
Comment (dktapps) [reply src/block/WoodLikeBlockIdHelper.php]: There is no bamboo log or "wood", but this isn't the way to fix it

Comment (AzaleeX) [reply src/block/WoodLikeBlockIdHelper.php]: So we're just going to create an id with an existing block?

**Comment by ShockedPlot7560** [comment]
Same problem as #5875 we can't introduce bamboo as a wood type

**Pull request closed without merging**